sforzando/q-lako
# Store Audit Logs

## Describe the PR

To close #11 .

Store the behavior of all users to CloudSQL.

## Screenshots

<img width="1539" alt="Screen Shot 2020-10-13 at 8 19 53" src="https://user-images.githubusercontent.com/32637762/95798534-3b07fa00-0d2d-11eb-8a3f-d8e41426b391.png">

## Detail of the change

- [x] Introduce [Google Secret Manager](https://cloud.google.com/secret-manager)
- [x] Introduce [Cloud SQL Proxy](https://cloud.google.com/sql/docs/mysql/sql-proxy)
- [x] ORM w/ [SQLAlchemy](https://www.sqlalchemy.org)
- [x] Add test codes for `audit_logger.py`
- [x] Implement GET `/audit_logs`
- [x] Introduce [Flask-MonitoringDashboard](https://flask-monitoringdashboard.readthedocs.io/en/latest/)

## Anticipated impacts

GitHub Actions now take longer due to the installation of `cloud_sql_proxy` and the addition of Secret Manager access.
See `.github/workflows/on_push.yml` .

## To reproduce

Steps to check the behavior:

1. Install cloud_sql_proxy `gcloud components install cloud_sql_proxy`
1. Launch cloud_sql_proxy `cloud_sql_proxy -instances=<CLOUD_SQL_CONNECTION_NAME>=tcp:3306`
1. Install dependencies `pip install --upgrade -r requirements.txt`
1. If you want to check [flask_monitoringdashboard](https://github.com/flask-dashboard/Flask-MonitoringDashboard) `pip install flask_monitoringdashboard`
1. Run `python main.py`

## Additional context

### .flake8

I Added `.flake8` to ignore **H306** and **H405**.
H306 forces libraries to be imported in alphabetical order, but is not compatible with PyCharm's import optimization.
H405 does not support [Google-style docstring](https://google.github.io/styleguide/pyguide.html).

### flask_monitoringdashboard

Currently, `flask_monitoringdashboard` is not compatible with the latest version of `Jinja2` (2.11.2), so it is excluded from `requirements.txt` .
